[DOCS] Adds text about data types and categorization to Anomaly Detection overview page (#809)
elastic/stack-docs: docs/en/stack/ml/anomaly-detection/categorization-data.asciidoc

[role="xpack"]
[[ml-datatypes-categorization]]
=== Data types and categorization

> Categorization is a {ml} process that observes the static parts of the data, 

Reviewer: ```suggestion
Categorization is a {ml} process that considers a tokenization of a field, 
```

> cluster similar data together, and classify them into categories. However, 

Reviewer 2: ```suggestion
clusters similar data together, and classifies them into categories. However, 
```

> categorization doesn't work equally efficient on different data types. It works 

Reviewer: ```suggestion
categorization doesn't work equally well on different data types. It works 
```

best on machine-written messages and application outputs, typically on data that 
consists of repeated elements, for example log messages for the purpose of 
system troubleshooting. Log categorization groups unstructured log messages into 
categories, then you can use {anomaly-detect} to model and identify rare or 
unusual counts of log message categories. For more information about the 
process, see 
{ml-docs}/ml-configuring-categories.html[Categorizing log messages].

The reason why categorization works best on data like log messages is that they 

> have structural similarities that can be recognized easily by the {ml} model. 

Reviewer: ```suggestion
Categorization is tuned to work best on data like log messages by taking token
order into account, not considering synonyms, and including stop words in its analysis.
```

Complete sentences in human communication or literary text (for example emails, 
wiki pages, prose, or other human generated content) can be extremely diverse in 

> structure, consequently categorization may provide poor results on such data. 

Reviewer: ```suggestion
structure.  Since categorization is tuned for machine data it will give poor results on such human generated data. 
```

For example, the categorization job would create so many categories that 

> couldn't be handled effectively.

Reviewer: ```suggestion
couldn't be handled effectively.  Categorization is _not_ NLP.
```

> couldn't be handled effectively.  Categorization is _not_ NLP.

Author: ```suggestion
couldn't be handled effectively.  Categorization is _not_ natural language processing (NLP).
```

[role="xpack"]
[[ml-datatypes-categorization]]
=== Data types and categorization

Categorization is a {ml} process that considers a tokenization of a field, 
clusters similar data together, and classifies them into categories. However, 
categorization doesn't work equally efficient on different data types. It works 
best on machine-written messages and application outputs, typically on data that 
consists of repeated elements, for example log messages for the purpose of 
system troubleshooting. Log categorization groups unstructured log messages into 
categories, then you can use {anomaly-detect} to model and identify rare or 
unusual counts of log message categories. For more information about the 
process, see 
{ml-docs}/ml-configuring-categories.html[Categorizing log messages].

> The reason why categorization works best on data like log messages is that they 

Author: ```suggestion

```
